core/src/com/unciv/models/ruleset/Ruleset.kt

```diff
@@ -297,13 +298,39 @@ class Ruleset {
         return stringList.joinToString { it.tr() }
     }
 
+    /** Similarity below which an untyped unique can be considered a potential misspelling.
+     * Roughly corresponds to the fraction of the Unique placeholder text that can be different/misspelled, but with some extra room for [getRelativeTextDistance] idiosyncrasies. */
+    private val uniqueMisspellingThreshold = 0.15 // Tweak as needed. Simple misspellings seem to be around 0.025, so would mostly be caught by 0.05. IMO 0.1 would be good, but raising to 0.15 also seemed to catch what may be an outdated Unique.
 
     fun checkUniques(uniqueContainer:IHasUniques, lines:RulesetErrorList,
                      severityToReport: UniqueType.UniqueComplianceErrorSeverity) {
         val name = if (uniqueContainer is INamed) uniqueContainer.name else ""
 
         for (unique in uniqueContainer.uniqueObjects) {
-            if (unique.type == null) continue
+            if (unique.type == null) {
```

Comment:
If this is your only check for whether a unique exists, that will probably give a lot of false positives until all uniques are typed. How much output does this give for the vanilla & G&K rulesets, including the 'unique looks like it should be fine, but for some reason isn't recognized'?
Additionally, we allow using uniques as a sort of 'identifier', similar to what we do with 'spaceship part'. Here we also catch these.

Reply:
See the screenshot. False positives are avoided by excluding anything that's *too* different. Basically, spelling similarity, not `unique.type == null`, is used as the main check for whether a unique exists, so the spelling has to be close enough to feasibly be a typo but not identical to count. If we start to see false positives, then we can just decrease `uniqueMisspellingThreshold` until they disappear. I think 0.1 would be good, but 0.05 would also work. Currently at 0.15, which is rather high, but based on the output in the screenshot it seems to still work fine.

No "Looks like it should be fine"s are currently generated for the stock rulesets. In in theory I don't think they should actually ever be generated, as the check for `null` I think should exclude all uniques that can produce them— I added it because I felt like I was going crazy trying to notice what was actually different about the spellings, so *not* seeing it affirms that there is indeed a spelling difference. A bug would have to be introduced that prevents `unique.type:UniqueType` from being set, I think, in order for it to actually ever show up.

Reply:
Ah, I misread the code for when that line is placed, my bad. The ones in the screenshot (after close inspection :upside_down_face:) are all indeed valid misspellings so I might have overreacted a bit

Reply:
Ah, no worries. My original notes for this PR had a heavy emphasis on false positives too, so I definitely think it's a valid concern— Hundreds of UniqueTypes with short strings made of just a couple stat and placeholder names, so you'd think some of them would have very similar names to untyped Uniques, right? But no, it just worked out better than either of us expected.